components/system-broker/pkg/http/secured_transport.go

```diff
@@ -61,3 +70,9 @@ func (c *SecuredTransport) refreshToken(ctx context.Context) error {
 
 	return nil
 }
+
+func (c *SecuredTransport) validToken() bool {
+	c.lock.RLock()
```

Comment:
which race condition do we solve here ? (e.g. if a token is being validated and its valid, no one will actually try to generate a new one)

Reply:
Discussed separately. In short this mitigates the issue that the token will be refreshed many times one after another because in the time when it is refreshed other gouroutines will check and see that it is expired and will wait to refresh also.